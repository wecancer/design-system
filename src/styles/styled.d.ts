import 'styled-components'
import theme from './theme'

type Theme = typeof theme

declare module 'styled-components' {
  // don't disable the eslint in other files
  /* eslint-disable @typescript-eslint/no-empty-interface */
  export interface DefaultTheme extends Theme {}
}
