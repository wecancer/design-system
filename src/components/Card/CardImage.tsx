import React, {useContext, useEffect} from 'react'
import styled, {css} from 'styled-components'
import Context from './Card.context'

const CardImageStyled = styled.div<{isLeft: boolean; heightSize: number; imageUrl: string}>`
  ${({imageUrl, heightSize, theme, isLeft}) => css`
    height: ${isLeft ? 78 : heightSize}px;
    font-family: ${theme.font.familyRedesign};
    background: url('${imageUrl}') no-repeat center;
    background-size: cover;
    position: relative;
    margin-bottom: -1rem;

    ${isLeft
      ? css`
          width: 78px;
          position: absolute;
          left: 0;
          top: 0;
          border-radius: 1rem;
          z-index: 3;
        `
      : css`
          border-radius: 1rem 1rem 0 0;
          z-index: 1;
        `}
  `}
`

type Props = React.HTMLAttributes<HTMLDivElement> & {
  position?: 'top-left'
  height?: number
  imageUrl: string
}

const CardImage = ({position, height = 100, ...props}: Props): React.ReactElement => {
  const {gapDirection, setGapDirection} = useContext(Context)

  useEffect(() => {
    setGapDirection(position === 'top-left' ? 'left' : 'none')
  }, [position, setGapDirection])

  return <CardImageStyled isLeft={gapDirection.includes('left')} heightSize={height} {...props} />
}

export default CardImage
