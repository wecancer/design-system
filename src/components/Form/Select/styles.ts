import {DefaultTheme} from 'styled-components'
import {GroupBase, StylesConfig} from 'react-select'

import {Option} from './types'

const selectStyles = ({font, colors}: DefaultTheme): StylesConfig<Option, true, GroupBase<Option>> => ({
  container: (styles) => ({
    ...styles,
    fontFamily: font.familyRedesign,
  }),
  option: (styles, {isSelected, isFocused}) => ({
    ...styles,
    padding: '0.375rem 1.5rem',
    ...(isSelected || isFocused
      ? {
          backgroundColor: colors.line,
          color: colors.titleActive,
        }
      : {}),
  }),
  menu: (styles) => ({
    ...styles,
    backgroundColor: colors.inputBg,
    padding: '0',
    borderRadius: '0.625rem',
    border: 'none',
    boxShadow: 'none',
  }),
  menuList: (styles) => ({
    ...styles,
    padding: '0.625rem 0',
  }),
  valueContainer: (styles) => ({
    ...styles,
    padding: '0 0.625rem',
  }),
  indicatorSeparator: () => ({
    display: 'none',
  }),
  indicatorsContainer: (styles) => ({
    ...styles,
    paddingRight: '0.75rem',
    '& > div': {
      color: colors.placeholder,
      transition: 'all 250ms ease',
      '&:hover': {
        color: colors.titleActive,
      },
    },
  }),
  control: (styles, state) => ({
    ...styles,
    height: '40px',
    borderRadius: '0.75rem',
    borderWidth: '0.125rem',
    ...(state.isFocused
      ? {
          color: colors.titleActive,
          borderColor: `${colors.primary} !important`,
          boxShadow: `0 0 0.062rem 0.375rem ${colors.focusPrimary}`,
          backgroundColor: colors.white,
        }
      : {borderColor: colors.inputBg, backgroundColor: colors.inputBg}),
    '&:hover': {
      borderColor: colors.inputBg,
    },
  }),
  multiValue: (styles) => ({
    ...styles,
    color: colors.offWhite,
    borderRadius: '1rem',
    backgroundColor: colors.primary,
    border: `2px solid ${colors.primary}`,
  }),
  multiValueLabel: (styles) => ({
    ...styles,
    fontSize: '0.81rem',
    fontWeight: 400,
    color: colors.offWhite,
    paddingLeft: '0.75rem',
    paddingRight: '0.375rem',
  }),
  multiValueRemove: (styles) => ({
    ...styles,
    borderRadius: '0.812rem',
    paddingRight: '4px',
    '&:hover': {
      color: colors.offWhite,
      background: 'rgba(0, 0, 0, 0.2)',
    },
  }),
})

export default selectStyles
