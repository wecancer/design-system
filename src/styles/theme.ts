export type ColorsTemplate =
  | 'primary'
  | 'secondary'
  | 'turquoise'
  | 'violet'
  | 'pink'
  | 'orange'
  | 'purple'
  | 'yellow'
  | 'magenta'
  | 'success'
  | 'warning'
  | 'error'
  | 'white'

export enum BgTypes {
  turquoise = 'bgTurquoise',
  violet = 'bgViolet',
  pink = 'bgPink',
  orange = 'bgOrange',
  success = 'bgSuccess',
  warning = 'bgWarning',
  error = 'bgError',
  input = 'inputBg',
}

const theme = {
  colors: {
    primary: '#0080EB',
    darkPrimary: '#00318A',
    lightPrimary: '#00BEF7',
    extraLightPrimary: '#8DE7E6',
    focusPrimary: '#CEE2FF',
    bgPrimary: '#D8E3FF',

    secondary: '#1A8F3F',
    darkSecondary: '#0D4C0B',
    lightSecondary: '#34C279',
    extraLightSecondary: '#7CEB93',
    bgSecondary: '#C7EBC6',

    titleActive: '#14142B',
    body: '#4E4B66',
    label: '#6E7191',
    placeholder: '#A0A3BD',
    line: '#D9DBE9',
    focusLine: '#F1F2F7',
    inputBg: '#EFF0F6',
    bg: '#F7F7FC',
    offWhite: '#FCFCFC',
    white: '#FFFFFF',

    turquoise: '#17C3B2',
    darkTurquoise: '#30ACAC',
    lightTurquoise: '#6CDEDE',
    bgTurquoise: '#6CDEDE',

    violet: '#7433FF',
    darkViolet: '#3D2EEA',
    lightViolet: '#624AF2',
    bgViolet: '#624AF2',

    pink: '#ED2E7E',
    darkPink: '#C30052',
    lightPink: '#FF84B7',
    bgPink: '#FF84B7',

    orange: '#FF7849ED',
    darkOrange: '#FF5216',
    lightOrange: '#FF9E7C',
    bgOrange: '#FFD8CB',

    purple: '#A62C9E',
    magenta: '#F51F4F',
    yellow: '#FF9B00',

    success: '#00BA88',
    darkSuccess: '#00966D',
    lightSuccess: '#F2FFFB',
    bgSuccess: '#F2FFFB',

    warning: '#F4B740',
    darkWarning: '#946200',
    lightWarning: '#FFD789',
    bgWarning: '#FFF4DF',

    error: '#DC3545',
    darkError: '#90131F',
    lightError: '#FF84B7',
    bgError: '#FFDFED',
  },
  boxShadow: {
    smooth: '0 4px 8px rgba(0, 0, 0, 0.06)',
    default: '0px 8px 16px rgba(17, 17, 17, 0.06)',
    sharp: '0px 8px 16px rgba(17, 17, 17, 0.4)',
  },
  border: {
    default: '1px solid ',
  },
  borderRadius: {
    default: '1rem',
  },
  container: {
    maxWidth: '1000px',
  },
  font: {
    lineHeight: 1.2,
    family: 'Roboto, sans-serif',
    familyRedesign: 'Poppins, sans-serif',
    weights: {
      light: '300',
      regular: '400',
      semiBold: '500',
      bold: '700',
    },
    sizes: {
      'display-1': '3rem',
      'display-2': '2.5rem',
      'display-3': '2rem',
      'display-4': '1.5rem',
      'display-5': '1.25rem',
      'display-6': '1rem',
      default: '1rem',
      small: '0.9rem',
      xsmall: '0.8rem',
    },
  },
}

export default theme
