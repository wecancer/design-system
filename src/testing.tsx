import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
