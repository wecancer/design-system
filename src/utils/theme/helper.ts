import {Appeareances, DefaultTheme, KeysOfAppearances} from 'styled-components'

export function getAppearance(appearances: Appeareances, appearance?: KeysOfAppearances) {
  if (!appearance) {
    return {
      main: '',
      darken: '',
      light: '',
    }
  }

  return appearances[appearance]
}

export function getBorderDefaultThemed(theme: DefaultTheme, appearance: KeysOfAppearances) {
  return `${theme.border.default} ${getAppearance(theme.appearances, appearance).main}`
}
