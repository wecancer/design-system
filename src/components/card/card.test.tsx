import React from 'react'

import Card from '.'
import CardIcon from './card-icon'
import CardImage from './card-image'
import CardValue from './card-value'
import CardContent from './card-content'
import { BgTypes } from '../../styles/theme'
import { render } from '../../testing'

describe('<Card />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <>
        <Card>
          <CardImage
            height={150}
            imageUrl="https://medicinasa.com.br/wp-content/uploads/2019/04/WeCancer2a.jpg"
          />
          <CardContent>The card with image</CardContent>
        </Card>
        <Card>
          <CardImage
            position="top-left"
            imageUrl="https://medicinasa.com.br/wp-content/uploads/2019/04/WeCancer2a.jpg"
          />
          <CardContent>
            <p>Item 1</p>
            <p>Item 2</p>
            <p>Item 3</p>
          </CardContent>
        </Card>
        <Card>
          <CardIcon name="check" size={32} />
          <CardContent>Card with icon</CardContent>
        </Card>
        <Card onClick={(e) => e.preventDefault()}>
          <CardContent>Card clickable</CardContent>
        </Card>
        <Card bgColor={BgTypes.error} onClick={(e) => e.preventDefault()}>
          <CardValue fillColor="error" value={10} label="Patients" />
          <CardContent>Card clickable</CardContent>
        </Card>
        <Card>
          <CardValue value="+99" label="Notes" />
          <CardContent>Check it out!</CardContent>
        </Card>
      </>,
    )
    expect(container).toMatchSnapshot()
  })
})
