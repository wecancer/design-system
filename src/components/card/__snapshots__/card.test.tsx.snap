// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render snapshot correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL gdXJKY"
  >
    <div
      class="sc-hKwDye bjuJie"
    />
    <div
      class="sc-iCfMLu dPfeOm"
    >
      The card with image
    </div>
  </div>
  <div
    class="sc-bdvvtL bumdSL"
  >
    <div
      class="sc-hKwDye fTarJn"
    />
    <div
      class="sc-iCfMLu hyKfPD"
    >
      <p>
        Item 1
      </p>
      <p>
        Item 2
      </p>
      <p>
        Item 3
      </p>
    </div>
  </div>
  <div
    class="sc-bdvvtL gdXJKY"
  >
    <div
      class="sc-dkPtRN eWGRQa"
    >
      <div
        class="sc-gsDKAQ jupgag"
        data-icon-name="check"
      >
        <svg
          fill="none"
          viewBox="0 0 18 13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 1L6 12L1 7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-iCfMLu dPfeOm"
    >
      Card with icon
    </div>
  </div>
  <div
    class="sc-bdvvtL gdXJKY"
    role="button"
    tabindex="0"
  >
    <div
      class="sc-iCfMLu dPfeOm"
    >
      Card clickable
    </div>
  </div>
  <div
    class="sc-bdvvtL bPDKIS"
    role="button"
    style="display: grid; grid-template-columns: minmax(0, 90px) 1fr; align-items: center;"
    tabindex="0"
  >
    <div
      class="sc-eCImPb hcFpqB"
    >
      <p
        class="sc-jRQBWg jSBYXt"
      >
         
        10
         
      </p>
      <p
        class="sc-gKclnd vZma-d"
      >
        Patients
      </p>
    </div>
    <div
      class="sc-iCfMLu KhiQc"
    >
      Card clickable
    </div>
  </div>
  <div
    class="sc-bdvvtL gdXJKY"
    style="display: grid; grid-template-columns: minmax(0, 90px) 1fr; align-items: center;"
  >
    <div
      class="sc-eCImPb imEwWK"
    >
      <p
        class="sc-jRQBWg jSBYXt"
      >
         
        +99
         
      </p>
      <p
        class="sc-gKclnd vZma-d"
      >
        Notes
      </p>
    </div>
    <div
      class="sc-iCfMLu dPfeOm"
    >
      Check it out!
    </div>
  </div>
</div>
`;
